webapi/src/main/scala/org/knora/webapi/messages/v2/responder/resourcemessages/ResourceMessagesV2.scala

```diff
@@ -75,9 +75,10 @@ case class ResourcesGetRequestV2(resourceIris: Seq[IRI],
   * Requests a preview of one or more resources. A successful response will be a [[ReadResourcesSequenceV2]].
   *
   * @param resourceIris   the IRIs of the resources to obtain a preview for.
+  * @param targetSchema   the schema of the response.
   * @param requestingUser the user making the request.
   */
-case class ResourcesPreviewGetRequestV2(resourceIris: Seq[IRI], requestingUser: UserADM) extends ResourcesResponderRequestV2
+case class ResourcesPreviewGetRequestV2(resourceIris: Seq[IRI], targetSchema: ApiV2Schema = ApiV2Complex, requestingUser: UserADM) extends ResourcesResponderRequestV2
```

Comment:
@benjamingeer I was wondering if we should remove the default value for the target schema. 

With `ApiV2Complex` as the default schema, list node labels are not queried which could lead to problem when trying to serialize the information in the simple schema.

Reply:
Yes, now I try to avoid default values, they're often a source of bugs.

Reply:
>  list node labels are not queried 

if no schema passed by accident

Reply:
@benjamingeer I propose we could look a this next week.

Reply:
OK, but is there something difficult about it?

Reply:
The only difficulty I see is internal use (checking updates etc.).

Reply:
All you have to do is remove the default value, and set the parameter explicitly to `ApiV2Complex` wherever it was previously set by default. Then everything will continue to work as before.

Reply:
Now that I think about it, `ResourcesPreviewGetRequestV2` doesn't query values at all, so actually I think this problem can never occur, right?

Reply:
I think the notion of what a schema is is changed with this and your previous standoff PR because at query time you now have to get more or less information.

Before, I just thought of schemas as something that only the Route has to know about.

Reply:
> ResourcesPreviewGetRequestV2 doesn't query values

no values, no problems ;-)

Reply:
> I think the notion of what a schema is is changed with this and your previous standoff PR because at query time you now have to get more or less information.

OK, but I don't see how that matters. If you remove a default value from a method, and change all the code that calls the method so that it passes the same value explicitly, you will get the same result. It doesn't matter what the method does.

> no values, no problems ;-)

Yes.

Reply:
> All you have to do is remove the default value, and set the parameter explicitly to ApiV2Complex wherever it was previously set by default. Then everything will continue to work as before.

done in https://github.com/dhlab-basel/Knora/pull/1321/commits/586762554438343d79cbd69a854dfb87e001dab4